rum(auto-instrumentation): add IBM HTTP Server (#27762)
DataDog/documentation: content/en/real_user_monitoring/browser/setup/server.md

Datadog RUM Server-Side Instrumentation (Auto-Instrumentation) lets you opt into Real User Monitoring (RUM) automatically by instrumenting web applications served through a web server or proxy.

RUM Auto-Instrumentation works by injecting the RUM Browser SDK into the HTML responses being served.

After your applications have been instrumented, you can configure your RUM application in Datadog.

## Limitations

Keep in mind the following limitations when using this setup:

- This instrumentation method **does not support [advanced RUM configurations][3]**, except for `allowedTracingUrls` and `excludedActivityUrls`.
- If your web server is acting as a proxy and the upstream server uses **end-to-end encryption (TLS)** or **content compression** (gzip, zstd, Brotli), the RUM Browser SDK may **not be injected**. To ensure proper instrumentation:
  - **Disable content compression** on the upstream server.
  - **Enable TLS origination** on the web server.

## Prerequisites

> The [Datadog Agent][2] is required. Make sure you already have installed and configured the Agent.

Reviewer: ```suggestion
The [Datadog Agent][2] is installed and configured.
```

{{% tab "NGINX" %}}

The Auto-Instrumentation method leverages the [NGINX Dynamic Modules capability][1] to implement a response body filter. The filter injects the RUM Browser SDK into the response body for responses
identified as HTML.

> For more granular control over instrumented RUM application, you can also manually install and configure the module.

Reviewer: ```suggestion
For more granular control over the instrumentation of a RUM application, you can also manually install and configure the module.
```

{{% /collapse-content %}}

{{% /tab %}}

> {{% tab "Apache HTTP Server / IBM HTTP Server" %}}

Reviewer: ```suggestion
{{% tab "Apache/IBM HTTP Server" %}}
```

The Auto-Instrumentation method leverages the [Apache httpd Modules capability][1] to implement a response body filter. The filter injects the RUM Browser SDK into the response body for responses
identified as HTML.

> For more granular control over instrumented RUM application, you can also manually install and configure the module.

Reviewer: ```suggestion
For more granular control over the instrumentation of the RUM application, you can also manually install and configure the module.
```

{{% /tab %}}

> {{% tab "Apache HTTP Server / IBM HTTP Server" %}}

Author: ```suggestion
{{% tab "Apache/IBM HTTP Server" %}}
```
